pkg/mount/switch_root_linux.go

```diff
@@ -126,32 +126,36 @@ func MoveMount(oldPath string, newPath string) error {
 // 'special' in this context refers to the following non-blockdevice backed
 // mounts that are almost always used: /dev, /proc, /sys, and /run.
 // This function will create the target directories, if necessary.
-// If the target directories already exists, they must be empty.
+// If the target directories already exist, they must be empty.
 // This function skips missing mounts.
 func AddSpecialMounts(newRoot string) error {
 	var mounts = []string{"/dev", "/proc", "/sys", "/run"}
 
 	for _, mount := range mounts {
 		path := filepath.Join(newRoot, mount)
-		// Skip all mounting if the directory does not exists.
+		// Skip all mounting if the directory does not exist.
 		if _, err := os.Stat(mount); os.IsNotExist(err) {
-			fmt.Println("switch_root: Skipping", mount)
+			log.Printf("switch_root: Skipping %s as the dir does not exist", mount)
 			continue
 		} else if err != nil {
 			return err
 		}
+		// Also skip if not currently a mount point
+		if same, err := SameFilesystem("/", mount); err != nil {
+			return err
+		} else if same {
+			log.Printf("switch_root: Skipping %s as it is not a mount", mount)
+			continue
+		}
 		// Make sure the target dir exists and is empty.
-		if _, err := os.Stat(path); os.IsNotExist(err) {
+		if fi, err := os.Stat(path); os.IsNotExist(err) {
 			if err := unix.Mkdir(path, 0); err != nil {
 				return err
 			}
 		} else if err != nil {
 			return err
-		}
-		if empty, err := DirIsEmpty(path); err != nil {
-			return err
-		} else if !empty {
-			return fmt.Errorf("%v must be empty", path)
+		} else if !fi.IsDir() {
+			return fmt.Errorf("%v must be a dir", path)
```

Comment:
might as well be consistent and use %q. 
you might want to consider redoing this just a bit to avoid all the else if bits. A common way would be
fi, err := os.Stat()
if err != nil && err != os.IsNotExist(err){
mkdir ...
}
and so on. No big deal, just the thought occurred, sometimes things go wrong with too much if then else

Reply:
I rebased on master, and force-pushed to get rid of the `Merge branch 'master' into switch_root_mounts` commit.

I added another commit addressing @rminnich 's comments.